### PR TITLE
Use 'table' format when possible

### PR DESCRIPTION
**Pull request by marco-c**
Ref #27

### CODE REVIEW

File: mozci/push.py [hunk 1]
```diff
@@ -155,7 +155,17 @@ def tasks(self):
...
+                data["data"] = [
+                    {
+                        field: entry[i]
+                        for i, field in enumerate(data["header"])
+                        if entry[i] is not None
+                    }
+                    for entry in data["data"]
+                ]
+
+            for task in data["data"]:
```
Comment (klahnakoski) [inline mozci/push.py]: rename `data` to `result`, if only of clarity

File: mozci/push.py [hunk 2]
```diff
@@ -155,7 +155,17 @@ def tasks(self):
...
         def add(data):
-            for task in data:
+            if "header" in data:
+                data["data"] = [
+                    {
+                        field: entry[i]
+                        for i, field in enumerate(data["header"])
+                        if entry[i] is not None
+                    }
+                    for entry in data["data"]
```
Comment (klahnakoski) [inline mozci/push.py]: use `zip` instead if an index (`i`):

```
[
    {
        field: value
        for field, value in zip(data['header'], entry) 
        if value is not None
    } 
    for entry in data['data'] 
]
```

or, if you do not care about extra Nones:

```
[dict(zip(data['header'], entry)) for entry in data['data']]
```

File: mozci/push.py (hunk 1 above)
Comment (marco-c) [reply mozci/push.py]: Done with b63faaf.

File: mozci/push.py (hunk 2 above)
Comment (marco-c) [reply mozci/push.py]: Done with 018f2b3.

**Pull request merged**